bin/faucet/src/faucet/mod.rs

```diff
@@ -38,6 +40,54 @@ mod store;
 
 pub const DISTRIBUTE_FUNGIBLE_ASSET_SCRIPT: &str = include_str!("distribute_fungible_asset.masm");
 
+// FAUCET PROVER
+// ================================================================================================
+
+/// Represents a transaction prover which can be either local or remote, and is used to prove
+/// transactions minted by the faucet.
+enum FaucetProver {
+    Local(LocalTransactionProver),
+    Remote(RemoteTransactionProver),
+}
+
+impl FaucetProver {
+    /// Creates a new local prover.
+    ///
+    /// It uses the default proving options.
+    fn local() -> Self {
+        Self::Local(LocalTransactionProver::new(ProvingOptions::default()))
+    }
+
+    /// Creates a new remote prover.
+    ///
+    /// # Arguments
+    ///
+    /// * `endpoint` - The endpoint to connect to the remote prover.
+    fn remote(endpoint: impl Into<String>) -> Self {
+        Self::Remote(RemoteTransactionProver::new(endpoint))
+    }
+
+    async fn prove(
+        &self,
+        tx: impl Into<TransactionWitness> + Clone,
+    ) -> Result<ProvenTransaction, MintError> {
+        match self {
+            Self::Local(prover) => prover.prove(tx.into()).await,
+            Self::Remote(prover) => {
+                let proven_tx = prover.prove(tx.clone().into()).await;
+                match proven_tx {
+                    Ok(proven_tx) => Ok(proven_tx),
+                    Err(err) => {
+                        error!("failed to prove transaction with remote prover, falling back to local prover: {}", err);
```

Comment:
I think we want to `warn` here instead of `error` since it's gracefully degrading